### PR TITLE
Implement tree structure in the project selector list #1953

### PR DESCRIPTION
**Pull request by ashklianko**

### CODE REVIEW

File: src/main/resources/assets/js/app/project/list/ProjectList.ts [hunk 1]
```diff
@@ -1,16 +1,90 @@
...
+            result.push(...this.unwrapProjectChildren(defaultProject));
+        }
+
+        const level0projects: Project[] = items.filter(
+            (project: Project) => !ProjectHelper.isDefault(project) && this.projectLevel.get(project.getName()) === 0);
+        level0projects.forEach((level0project: Project) => {
+            result.push(level0project);
+            result.push(...this.unwrapProjectChildren(level0project));
+        });
+
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/project/list/ProjectList.ts:75]: Hmm... Why do we have a special check for `defaultProject` and exclude if from `level0projects`  if we do the same operation (`result.push`, `this.unwrapProjectChildren`) for both?

File: src/main/resources/assets/js/app/project/list/ProjectList.ts [hunk 2]
```diff
@@ -1,16 +1,90 @@
...
+        return result;
+    }
+
+    private unwrapProjectChildren(project: Project): Project[] {
+        const result: Project[] = [];
+
+        this.projectChildren.get(project.getName()).forEach((child: Project) => {
+            result.push(child);
+            result.push(...this.unwrapProjectChildren(child));
```
Comment (alansemenov) [inline src/main/resources/assets/js/app/project/list/ProjectList.ts:84]: This piece of code is repeated three times now, one here and two in `getDirectProjectChildren` (should be once there, as I wrote above). But it seems like it deserves its own method.

File: src/main/resources/assets/js/app/project/list/ProjectList.ts (hunk 1 above)
Comment (ashklianko) [reply src/main/resources/assets/js/app/project/list/ProjectList.ts:75]: To ensure that Default project will always be first on the projects list

Comment (alansemenov) [reply src/main/resources/assets/js/app/project/list/ProjectList.ts:75]: Ok, then we can just sort `result` before returning it, to make sure the default project is first.

Comment (ashklianko) [reply src/main/resources/assets/js/app/project/list/ProjectList.ts:75]: We need to have default's chidren projects (with their children etc.) to be in the list right after default entry, so it is better this way - first process default with children, then the rest

**Comment by ashklianko** [comment]
Some issue with github, had to create new PR https://github.com/enonic/app-contentstudio/pull/2007

**Pull request merged**